scripts/ci/validate-metadata-files.js

```diff
@@ -0,0 +1,184 @@
+	const packagesData = await Promise.all( packagesPaths.map( getPackageData ) );
+	const packagesWithMetadata = packagesData.filter( p => p.metadata );
+
+	const invalidMetadata = [];
+
+	for ( const packageData of packagesWithMetadata ) {
+		const missingExports = getMissingExports( packageData );
+		const missingIcons = await getMissingIcons( packageData, options );
+
+		if ( ![ ...missingExports, ...missingIcons ].length ) {
+			continue;
+		}
+
+		invalidMetadata.push( {
+			name: packageData.name,
+			missingExports,
+			missingIcons
+		} );
+	}
+
+	if ( !invalidMetadata.length ) {
+		console.log( green( 'Validation successful.' ) );
+
+		return;
+	}
+
+	for ( const x of invalidMetadata ) {
+		console.log( red( `\nPackage "${ x.name }" has invalid metadata entries:` ) );
+
+		if ( x.missingExports.length ) {
+			console.log( red( [
+				'Missing exports:',
+				...x.missingExports.map( missingExport => ` - ${ missingExport }` )
+			].join( '\n' ) ) );
+		}
+
+		if ( x.missingIcons.length ) {
+			console.log( red( [
+				'Missing icons:',
+				...x.missingIcons.map( missingIcon => ` - ${ missingIcon }` )
+			].join( '\n' ) ) );
+		}
+	}
+
+	process.exit( 1 );
+}
+
+/**
+ * Gets plugins from metadata file at provided path.
+ * Additionally, adds `packageName` of the plugin to returned data.
+ */
+async function getPackageData( packagePath ) {
+	const name = packagePath.split( '/' ).at( -1 );
+
+	const metadataPath = upath.join( packagePath, 'ckeditor5-metadata.json' );
+	const indexPath = upath.join( packagePath, 'src', 'index.ts' );
+
+	let metadata = null;
+	let index = null;
+
+	if ( await fs.exists( metadataPath ) ) {
+		metadata = await fs.readJSON( metadataPath );
+	}
+
+	if ( await fs.exists( indexPath ) ) {
+		index = await fs.readFile( indexPath, 'utf-8' );
+	}
+
+	return { name, metadata, index };
+}
+
+function getMissingExports( packageData ) {
+	if ( !packageData.index ) {
+		throw new Error( 'missing index' );
+	}
+
+	const requiredExports = packageData.metadata.plugins.map( plugin => CUSTOM_EXPORT_NAMES[ plugin.className ] || plugin.className );
+
+	const ast = parser.parse( packageData.index, {
+		sourceType: 'module',
+		ranges: true,
+		plugins: [ 'typescript' ]
+	} );
+
+	const exports = ast.program.body
+		.filter( node => node.type.startsWith( 'Export' ) )
+		.flatMap( node => {
+			if ( node.specifiers ) {
+				return node.specifiers.map( specifier => specifier.exported.loc.identifierName );
+			}
+		} );
+
+	return requiredExports.filter( requiredExport => !exports.includes( requiredExport ) );
+}
+
+async function getMissingIcons( packageData, options ) {
+	const missingIcons = [];
+
+	for ( const pluginData of packageData.metadata.plugins ) {
+		if ( !pluginData.uiComponents ) {
+			continue;
+		}
+
+		const iconPaths = pluginData.uiComponents.map( uiComponent => uiComponent.iconPath ).filter( Boolean );
+
+		if ( !iconPaths.length ) {
+			continue;
+		}
```

Comment:
```suggestion
	const pluginsToCheck = packageData.metadata.plugins
		.filter( ( { uiComponents } ) ) => !uiComponents )
		.filter( ( { uiComponents } ) => {
			return uiComponents.map( ( { iconPath } ) => iconPath )
				.filter( Boolean )
				.length;
		} );

	for ( const pluginData of pluginsToCheck ) {
```

This would simplify the `for..of` loop. This loop could look even nicer after simplifying the `try/catch`. Not to mention that it's easier to modify and understand what's going on here.